### DISCUSSION TITLE
Add support for graalvm style Java.type() function

### DISCUSSION THREAD

**Discussion by gordhosted** [caoccao/Javet#84] [Ideas]
The following seems to work and would allow a script to create any class with the same syntax as graalvm with Java.type(String s). Then you don't have to know ahead of time which classes to inject into the v8Runtime global object.

Are there any issues anyone can see with this approach?

index.js
```
var Java = new JavaClass();
var FileClass = Java.type('java.io.File');
var test = new FileClass("myFile.md");

console.log(test.getName());
```
main.java
```
try (V8Runtime v8Runtime = V8Host.getNodeInstance().createV8Runtime())
{
	v8Runtime.setConverter(new JavetProxyConverter());

	v8Runtime.getGlobalObject().set("JavaClass", Java.class);

	File script = new File("./index.js");
	String s = FileUtils.readFileToString(script);

	v8Runtime.getExecutor(s).executeVoid();
			
	v8Runtime.getGlobalObject().delete("JavaClass");
}
```
Java.java
```
public class Java
{
	public Class type(String s)
	{
		try
		{
			return Class.forName(s);
		}
		catch (ClassNotFoundException e)
		{
			e.printStackTrace();
		}
		
		return null;
	}
}

```

**Comment by gordhosted** [comment]
just saw @caoccao latest post on a different issue https://github.com/oracle/graaljs/issues/2#issuecomment-907315617 and it addresses the above:

> Javet comes with a different mindset from what GraalVM defines. In Javet, you may inject java.lang.Class.class in Node.js, then simply call Class.forName(...) in Node.js to load whatever you want to load. That's a native solution which allows you to even specify the classloader. Following the same way, you may inject File.class or whatever classes. Please refer to this doc for more details.

so that seems to answer it, no problems just the obvious security implications

**Comment by caoccao** [comment]
Congratulations. I'm glad that you figured out how to simply resolve that.

Javet gives you all kind of ways on defining your own DSL, even simulating GraalJS. That's also the general guideline in Javet design: Javet avoids defining an official way of how to inject Java objects, instead Javet teaches people how to define their own ways in a few lines of code.
